app/src/main/java/org/stepic/droid/ui/activities/LaunchActivity.kt

```diff
@@ -247,20 +248,19 @@ class LaunchActivity : SmartLockActivityBase(), LoginView {
             val result = Auth.GoogleSignInApi.getSignInResultFromIntent(data)
             // here is not only fail due to Internet, fix it. see: https://developers.google.com/android/reference/com/google/android/gms/auth/api/signin/GoogleSignInResult
             if (result.isSuccess) {
-                val account = result.signInAccount
-                if (account == null) {
-                    onInternetProblems()
-                    return
-                }
-                val authCode = account.serverAuthCode
+                val authCode = result?.signInAccount?.serverAuthCode
```

Comment:
In if statement `result` is considered as not-null value but on the next line, it's checked for null.